libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +82,193 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
+    } else {
+      this.handleOnKeyForClosedMenu(event);
+    }
+  }
+
+  private preventFurtherPropagation(event: KeyboardEvent) {
+    event.stopImmediatePropagation();
+    event.preventDefault();
+  }
+
+  private getFirstInteractiveElement(el: HTMLIonItemElement) {
+    const controls = el.querySelectorAll<
+      HTMLIonToggleElement | HTMLIonRadioElement | HTMLIonCheckboxElement
+    >('ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled])');
+
+    return controls[0] || undefined;
+  }
+
+  private handleOnKeyForClosedMenu(event: KeyboardEvent) {
+    const key = event.key;
+    switch (key) {
+      case ' ':
+      case 'Enter':
+      case 'ArrowDown':
+      case 'Down':
+        this.focusedIndex = 0;
+        this._floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Up':
+      case 'ArrowUp':
+        this.focusedIndex = this.kirbyItems.length - 1;
+        this._floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+    }
+  }
+
+  private handleOnKeyForOpenedMenu(event: KeyboardEvent) {
+    const key = event.key;
+    switch (key) {
+      case 'ArrowDown':
+        if (this.focusedIndex < this.kirbyItems.length - 1) {
+          this.focusedIndex++;
+          this.focusItem();
+          this.preventFurtherPropagation(event);
+        }
+        break;
+      case 'ArrowUp':
+        if (this.focusedIndex > 0) {
+          this.focusedIndex--;
+          this.focusItem();
+          this.preventFurtherPropagation(event);
+        }
+        break;
+      case 'Escape':
+        if (this.closeOnEscapeKey) {
+          this._floatingMenu.hide();
+        }
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Tab':
+        this._floatingMenu.hide();
+        break;
+    }
+  }
+
+  private focusTriggerButton() {
+    this.getTriggerButton().nativeElement.focus();
+  }
+
+  resetFocus() {
+    this.focusedIndex = -1;
+  }
+
+  focusItem() {
+    const itemToBeFocused = this.kirbyItems.get(this.focusedIndex);
+    const ionItem = itemToBeFocused.nativeElement.querySelector('ion-item');
+
+    // Look for interactive element within ion-item like toggle or checkbox and set focus if found
+    const firstInteractiveElementWithinItem = this.getFirstInteractiveElement(ionItem);
+    if (firstInteractiveElementWithinItem) {
+      firstInteractiveElementWithinItem['setFocus']();
+    } else {
+      this.focusSelectableItem(ionItem);
+    }
+  }
+
+  private focusSelectableItem(ionItem: HTMLIonItemElement) {
+    const selectableItem: HTMLButtonElement =
+      ionItem.shadowRoot.querySelector('button:not([disabled])');
+    if (selectableItem) {
+      selectableItem.focus();
+    }
+  }
+
+  getTriggerButton(): ElementRef<HTMLButtonElement> {
+    return this.userProvidedButton ?? this.defaultButtonElement;
+  }
```

Comment:
Consider returning the native element to simplify usage:
```suggestion
  getTriggerButton(): HTMLButtonElement {
    return (this.userProvidedButton ?? this.defaultButtonElement).nativeElement;
  }
```